src/constants.ts

```diff
@@ -330,7 +331,30 @@ export const ALLOWED_FORMATTING_TAGS = ['b', 'i', 'u', 'em', 'strong', 'span']
 export const ALLOWED_TAGS = ['ul', 'li', 'br', ...ALLOWED_FORMATTING_TAGS]
 
 export const ALLOWED_ATTRIBUTES = {
-  span: ['class', 'style']
+  span: ['class', 'style'],
 }
 
 export const EMPTY_SPACE = '  '
+
+export enum SHORTCUTS_ID {
+  // disable undo/redo until deepClone is optimized
+  // 'undo',
+  // 'redo',
+  // 'search',
+  OUTDENT = 'outdent',
+  INDENT = 'indent',
+  TOGGLE_TABLE_VIEW = 'toggleTableView',
+  TOGGLE_SORT = 'toggleSort',
+  PIN_OPEN = 'pinOpen',
+  PIN_SUB_THOUGHTS = 'pinSubthoughts',
+  NOTE = 'note',
+  DELETE = 'delete',
+  TOGGLE_CONTEXT_VIEW = 'toggleContextView',
+  PROSE_VIEW = 'proseView',
+  // TOGGLESPLITVIEW = // 'toggleSplitView',
+  SPLIT_SENTENCES = 'splitSentences',
+  SUBCATEGORIZE_ONE = 'subcategorizeOne',
+  SUBCATEGORIZE_ALL = 'subcategorizeAll',
+  TOGGLE_HIDDEN_THOUGHTS = 'toggleHiddenThoughts',
+  EXPORT_CONTEXT = 'exportContext',
+}
```

Comment:
This would need to be manually maintained in parallel with the shortcuts index. If you want type safe shortcut ids without the overhead of a parallel enum, you can derive the type from the keys of https://github.com/cybersemics/em/blob/dev/src/shortcuts/index.ts.

Case in point, this list includes just the default shortcuts, not the full list of shortcuts.

(I'm aware that there are other instances in this file that do not follow this advice.)

Reply:
@raineorshine, the derived keys don't match the ids. ie.: the delete is named `deleteEmptyThoughtOrOutdent` and the id is `delete`.

Reply:
They are two separate shortcuts. The id `delete` should be called `archive`.

https://github.com/cybersemics/em/blob/db627f09a57d5def2a053098978b12bdd4b9c696/src/shortcuts/archive.tsx#L69-L78

Reply:
Fixed, by creating the native archive component.